core/src/test/scala/com/amadeus/dataio/pipes/kafka/streaming/KafkaOutputTest.scala

```diff
@@ -56,13 +60,137 @@ class KafkaOutputTest extends AnyWordSpec with Matchers {
+      )
+      kafkaStreamOutput.trigger shouldEqual Some(Trigger.Continuous(Duration("60 seconds")))
+    }
+
+    "be initialized with trigger AvailableNow" in {
+
+      val config = ConfigFactory.parseMap(
+        Map(
+          "Output" -> Map(
+            "Type"    -> "com.amadeus.dataio.output.streaming.KafkaOutput",
+            "Brokers" -> "bktv001:9000, bktv002.amadeus.net:8000",
+            "Topic"   -> "test.topic",
+            "Mode"    -> "append",
+            "Trigger" -> "AvailableNow",
+            "Timeout" -> "24",
+            "Options" -> Map(
+              "failOnDataLoss"                       -> "false",
+              "maxOffsetsPerTrigger"                 -> "20000000",
+              "\"kafka.security.protocol\""          -> "SASL_PLAINTEXT",
+              "\"kafka.sasl.kerberos.service.name\"" -> "kafka"
+            )
+          )
+        )
+      )
+
+      val kafkaStreamOutput = KafkaOutput(config.getConfig("Output"))
+
+      kafkaStreamOutput.outputName shouldEqual None
+      kafkaStreamOutput.brokers shouldEqual "bktv001:9000, bktv002.amadeus.net:8000"
+      kafkaStreamOutput.topic shouldEqual "test.topic"
+      kafkaStreamOutput.options shouldEqual Map(
+        "failOnDataLoss"                   -> "false",
+        "maxOffsetsPerTrigger"             -> "20000000",
+        "kafka.security.protocol"          -> "SASL_PLAINTEXT",
+        "kafka.sasl.kerberos.service.name" -> "kafka"
+      )
+      kafkaStreamOutput.trigger shouldEqual Some(Trigger.AvailableNow())
+    }
+
+    "be initialized with None trigger" in {
+
+      val config = ConfigFactory.parseMap(
+        Map(
+          "Output" -> Map(
+            "Type"    -> "com.amadeus.dataio.output.streaming.KafkaOutput",
+            "Brokers" -> "bktv001:9000, bktv002.amadeus.net:8000",
+            "Topic"   -> "test.topic",
+            "Mode"    -> "append",
+            "Timeout" -> "24",
+            "Options" -> Map(
+              "failOnDataLoss"                       -> "false",
+              "maxOffsetsPerTrigger"                 -> "20000000",
+              "\"kafka.security.protocol\""          -> "SASL_PLAINTEXT",
+              "\"kafka.sasl.kerberos.service.name\"" -> "kafka"
+            )
+          )
+        )
+      )
+
+      val kafkaStreamOutput = KafkaOutput(config.getConfig("Output"))
+
+      kafkaStreamOutput.outputName shouldEqual None
+      kafkaStreamOutput.brokers shouldEqual "bktv001:9000, bktv002.amadeus.net:8000"
+      kafkaStreamOutput.topic shouldEqual "test.topic"
       kafkaStreamOutput.options shouldEqual Map(
         "failOnDataLoss"                   -> "false",
         "maxOffsetsPerTrigger"             -> "20000000",
         "kafka.security.protocol"          -> "SASL_PLAINTEXT",
         "kafka.sasl.kerberos.service.name" -> "kafka"
       )
+      kafkaStreamOutput.trigger shouldEqual None
+    }
+
+    "throw an exception given missing topic but pattern" in {
```

Comment:
Based on the comment in the KafkaOutput class, this test and the following one need to be removed.